WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/minified/MinifiedWidgetUpdater.kt

```diff
@@ -0,0 +1,120 @@
+package org.wordpress.android.ui.stats.refresh.lists.widget.minified
+
+import android.appwidget.AppWidgetManager
+import android.content.ComponentName
+import android.content.Context
+import android.widget.RemoteViews
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.fluxc.store.stats.insights.TodayInsightsStore
+import org.wordpress.android.ui.prefs.AppPrefsWrapper
+import org.wordpress.android.ui.stats.StatsTimeframe
+import org.wordpress.android.ui.stats.refresh.lists.widget.WidgetUpdater
+import org.wordpress.android.ui.stats.refresh.lists.widget.configuration.StatsColorSelectionViewModel.Color.DARK
+import org.wordpress.android.ui.stats.refresh.lists.widget.configuration.StatsColorSelectionViewModel.Color.LIGHT
+import org.wordpress.android.ui.stats.refresh.lists.widget.configuration.StatsDataTypeSelectionViewModel.DataType
+import org.wordpress.android.ui.stats.refresh.lists.widget.configuration.StatsDataTypeSelectionViewModel.DataType.COMMENTS
+import org.wordpress.android.ui.stats.refresh.lists.widget.configuration.StatsDataTypeSelectionViewModel.DataType.LIKES
+import org.wordpress.android.ui.stats.refresh.lists.widget.configuration.StatsDataTypeSelectionViewModel.DataType.VIEWS
+import org.wordpress.android.ui.stats.refresh.lists.widget.configuration.StatsDataTypeSelectionViewModel.DataType.VISITORS
+import org.wordpress.android.ui.stats.refresh.lists.widget.utils.WidgetUtils
+import org.wordpress.android.ui.stats.refresh.utils.MILLION
+import org.wordpress.android.ui.stats.refresh.utils.ONE_THOUSAND
+import org.wordpress.android.ui.stats.refresh.utils.toFormattedString
+import org.wordpress.android.util.NetworkUtilsWrapper
+import org.wordpress.android.viewmodel.ResourceProvider
+import javax.inject.Inject
+
+class MinifiedWidgetUpdater
+@Inject constructor(
+    private val appPrefsWrapper: AppPrefsWrapper,
+    private val siteStore: SiteStore,
+    private val networkUtilsWrapper: NetworkUtilsWrapper,
+    private val resourceProvider: ResourceProvider,
+    private val todayInsightsStore: TodayInsightsStore,
+    private val widgetUtils: WidgetUtils
+) : WidgetUpdater {
+    override fun updateAppWidget(
+        context: Context,
+        appWidgetManager: AppWidgetManager,
+        appWidgetId: Int
+    ) {
+        val wideView = widgetUtils.isWidgetWiderThanLimit(appWidgetManager, appWidgetId)
```

Comment:
Thinking about this, maybe we should name this variable something like `isWideView` (here and in other places), since it's not actually a view :) WDYT?

Reply:
true, I often not use `is` cause the variable is a boolean but you're right, it's consistent and I should use it 👍